src/Symfony/Component/Finder/Finder.php

```diff
@@ -685,6 +706,14 @@ private function searchInDirectory(string $dir): \Iterator
             $notPaths[] = '#(^|/)\..+(/|$)#';
         }
 
+        if (static::IGNORE_VCS_IGNORED_FILES === (static::IGNORE_VCS_IGNORED_FILES & $this->ignore)) {
+            $gitIgnoreFilePath = sprintf('%s/.gitignore', $dir);
+            if (!is_readable($gitIgnoreFilePath)) {
+                throw new \RuntimeException(sprintf('File "%s" is not readable, can not use ignoreVCSIgnored.', $gitIgnoreFilePath));
```

Comment:
What about `The "ignoreVCSIgnored" option cannot be used by the Finder as the "%s" file is not readable.`.